trustfall_core/src/schema/mod.rs

```diff
@@ -397,11 +402,11 @@ fn check_type_and_property_and_edge_invariants(
 
                     // Check that the edge field doesn't have
                     // a list-of-list or more nested list type.
-                    match &field_type.base {
-                        BaseType::Named(_) => {}
-                        BaseType::List(inner) => match &inner.base {
-                            BaseType::Named(_) => {}
-                            BaseType::List(_) => {
+                    match &field_type.value() {
```

Comment:
this can also be expressed as
```rs
if matches!(&field_type.value(), InnerType::ListInnerType(ty) if matches!(ty.value(), InnerType::ListInnerType(_))) {
```

but I don't think it's better, and it's out of scope